build.gradle

```diff
@@ -54,6 +57,25 @@ allprojects {
     configFile = new File(rootDir, "checkstyle/checkstyle.xml")
   }
 
+  task createConfigDocDir() {
+    outputs.dir configDocDir
+    file(configDocDir).mkdirs()
+  }
+
+  task createConfigDocs( dependsOn : ['createConfigDocDir', 'compileJava'], type : JavaExec) {
+    classpath sourceSets.main.runtimeClasspath
+    main = 'com.linkedin.kmf.common.ConfigDocumentationGenerator'
+    args = [configDocDir,
```

Comment:
The code itself does not really depend the functionality of the input classes so adding them as compile time dependencies does not really make any sense.  If the list is wrong the build will still fail.